### PR TITLE
Improve support for new PHP8.2 standalone types

### PR DESCRIPTION
**Pull request by Wirone**
Appendix to FriendsOfPHP/PHP-CS-Fixer#6427

- Add support for `true` type
- Use types also in functions' params

### CODE REVIEW

File: src/Fixer/Casing/NativeFunctionTypeDeclarationCasingFixer.php [hunk 1]
```diff
@@ -87,6 +88,10 @@ public function __construct()
             $this->hints = array_merge($this->hints, ['never' => true]);
         }
 
+        if (\PHP_VERSION_ID >= 80200) {
+            $this->hints = array_merge($this->hints, ['true' => true]);
```
Comment (kubawerlos) [inline src/Fixer/Casing/NativeFunctionTypeDeclarationCasingFixer.php:92]:
```suggestion
            $this->hints['true'] = true;
```

File: tests/Fixer/Casing/NativeFunctionTypeDeclarationCasingFixerTest.php [hunk 2]
```diff
@@ -211,16 +211,21 @@ public function testFix82(string $expected, string $input): void
...
-            '<?php class T { public function Foo(object $A): null {return null;}}',
-            '<?php class T { public function Foo(object $A): NULL {return null;}}',
+        yield 'standalone type `false`' => [
+            '<?php class T { public function Foo(false $A): false {return $A;}}',
+            '<?php class T { public function Foo(False $A): FALSE {return $A;}}',
+        ];
+
+        yield 'standalone type `null`' => [
+            '<?php class T { public function Foo(null $A): null {return $A;}}',
+            '<?php class T { public function Foo(Null $A): NULL {return $A;}}',
```
Comment (kubawerlos) [inline tests/Fixer/Casing/NativeFunctionTypeDeclarationCasingFixerTest.php:228]: Can you add one more test with `\null`, `\true` and `\false`?

**Review by kubawerlos — CHANGES_REQUESTED** [review]

File: tests/Fixer/Casing/NativeFunctionTypeDeclarationCasingFixerTest.php (hunk 2 above)
Comment (Wirone) [reply tests/Fixer/Casing/NativeFunctionTypeDeclarationCasingFixerTest.php:228]: Since we have 2 usages of type in each test, we can use non-backslashed and backslashed version, without adding another test. Would it be acceptable?

Comment (kubawerlos) [reply tests/Fixer/Casing/NativeFunctionTypeDeclarationCasingFixerTest.php:228]: It should be fine, but I see now we have `fixFunctionArgumentTypes` and `fixFunctionReturnType` methods in that fixer, so let's keep param `$A` and return type as it is and add param `$B` with backslashes type, WDYT?

**Comment by SpacePossum** [comment]
Thanks @Wirone :+1:  I will work on this branch a bit more :)
